[2024-11-05T22:12:15.619+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_transform.transform manual__2024-11-05T22:12:14.932566+00:00 [queued]>
[2024-11-05T22:12:15.628+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_transform.transform manual__2024-11-05T22:12:14.932566+00:00 [queued]>
[2024-11-05T22:12:15.629+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-11-05T22:12:15.655+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): transform> on 2024-11-05 22:12:14.932566+00:00
[2024-11-05T22:12:15.663+0000] {standard_task_runner.py:57} INFO - Started process 1775 to run task
[2024-11-05T22:12:15.666+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_transform', 'transform', 'manual__2024-11-05T22:12:14.932566+00:00', '--job-id', '62', '--raw', '--subdir', 'DAGS_FOLDER/spark_transform.py', '--cfg-path', '/tmp/tmpmezjmc1g']
[2024-11-05T22:12:15.667+0000] {standard_task_runner.py:85} INFO - Job 62: Subtask transform
[2024-11-05T22:12:15.724+0000] {task_command.py:416} INFO - Running <TaskInstance: spark_transform.transform manual__2024-11-05T22:12:14.932566+00:00 [running]> on host c5156f201d33
[2024-11-05T22:12:15.810+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_transform' AIRFLOW_CTX_TASK_ID='transform' AIRFLOW_CTX_EXECUTION_DATE='2024-11-05T22:12:14.932566+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-11-05T22:12:14.932566+00:00'
[2024-11-05T22:12:15.823+0000] {base.py:73} INFO - Using connection ID 'SPARK_CONNECTION' for task execution.
[2024-11-05T22:12:15.824+0000] {spark_submit.py:366} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name arrow-spark jobs/example.py
[2024-11-05T22:12:15.927+0000] {spark_submit.py:536} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-11-05T22:12:15.930+0000] {spark_submit.py:536} INFO - JAVA_HOME is not set
[2024-11-05T22:12:15.944+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 467, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark jobs/example.py. Error code is: 1.
[2024-11-05T22:12:15.952+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=spark_transform, task_id=transform, execution_date=20241105T221214, start_date=20241105T221215, end_date=20241105T221215
[2024-11-05T22:12:15.969+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 62 for task transform (Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark jobs/example.py. Error code is: 1.; 1775)
[2024-11-05T22:12:16.001+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-11-05T22:12:16.040+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
